code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py

```diff
@@ -0,0 +1,55 @@
+# PROBLEM STATEMENT-> You are given an array A of size N and Q queries.(All indexes are 1 based).
+# Each query can be of two types
+#       1->   1 i  - add 1 to the ith element of the array A
+#       2->   2 K  - print the Kth odd number of the array A if it exists, else print -1
+
+import math
+n, q = list(map(int, input().strip().split(' ')))
+array = list(map(int, input().strip().split(' ')))
+
+
+buckets = int(math.ceil(math.sqrt(n)))
+total_odds = [0 for i in range(buckets)]
+
+bucket_size = int(math.ceil(n / buckets))
+
+
+
+for bucket in range(buckets):
```

Comment:
Please put all of the algorithm code itself in its own function.

Reply:
OK sir .

Reply:
done sir! please review my latest PR
 https://github.com/OpenGenus/cosmos/pull/4663